src/components/Proposal/Create/PluginForms/CreateTokenTradeProposal.tsx

```diff
@@ -0,0 +1,325 @@
+                <Field
+                  id="urlInput"
+                  maxLength={120}
+                  placeholder="Description URL"
+                  name="url"
+                  type="text"
+                  className={touched.url && errors.url ? css.error : null}
+                />
+
+                <div className={css.reward}>
+                  <label htmlFor="sendTokenAmountInput">
+                    Send token amount
+                    <ErrorMessage name="sendTokenAmount">{(msg) => <span className={css.errorMessage}>{msg}</span>}</ErrorMessage>
+                  </label>
+                  <div className={css.externalTokenInput}>
+                    <div className={css.amount}>
+                      <Field
+                        id="sendTokenAmountInput"
+                        placeholder={"How many tokens to send"}
+                        name="sendTokenAmount"
+                        type="number"
+                        className={touched.sendTokenAmount && errors.sendTokenAmount ? css.error : null}
+                        min={0}
+                        step={1}
+                      />
+                    </div>
+                    <div className={css.select}>
+                      <Field
+                        id="sendTokenAddress"
+                        name="sendTokenAddress"
+                        component={SelectField}
+                        options={Object.keys(supportedTokens()).map((tokenAddress) => {
+                          const token = supportedTokens()[tokenAddress];
+                          return { value: tokenAddress, label: token["symbol"] };
+                        })}
+                      />
+                    </div>
+                  </div>
+                </div>
+
+                <div className={css.reward}>
+                  <label htmlFor="receiveTokenAmountInput">
+                    Receive token amount
+                    <ErrorMessage name="receiveTokenAmount">{(msg) => <span className={css.errorMessage}>{msg}</span>}</ErrorMessage>
+                  </label>
+                  <div className={css.externalTokenInput}>
+                    <div className={css.amount}>
+                      <Field
+                        id="receiveTokenAmountInput"
+                        placeholder={"How many tokens to receive"}
+                        name="receiveTokenAmount"
+                        type="number"
+                        className={touched.receiveTokenAmount && errors.receiveTokenAmount ? css.error : null}
+                        min={0}
+                        step={1}
+                      />
+                    </div>
+                    <div className={css.select}>
+                      <Field
+                        id="receiveTokenAddress"
+                        name="receiveTokenAddress"
+                        component={SelectField}
+                        options={Object.keys(supportedTokens()).map((tokenAddress) => {
```

Comment:
To reduce the number of renders, this function should defined on the class, not here.  Note it only needs to be executed once, so you can use it to create a class array property member during construction. Reference the property member here.